### PR TITLE
Deprecate usecase scaladoc tag

### PR DESCRIPTION
**Pull request by m-sp**
Fixes https://github.com/scala/scala-dev/issues/584

### CODE REVIEW

**Comment by SethTisue** [comment]
cool! can I interest you in also removing all the places in this repo where it is still used...?

**Comment by m-sp** [comment]
I'm trying to get around to it on friday!

**Review by lrytz — APPROVED** [review]
👍 👍

File: src/library/scala/collection/immutable/List.scala [hunk 1]
```diff
@@ -94,12 +94,8 @@ sealed abstract class List[+A]
...
     *  @return  a list which contains `x` as first element and
     *           which continues with this list.
-    *
-    *  @usecase def ::(elem: A): List[A]
-    *    @inheritdoc
-    *
-    *    Example:
-    *    {{{1 :: List(2, 3) = List(2, 3).::(1) = List(1, 2, 3)}}}
+    *  Example:
+    *  {{{1 :: List(2, 3) = List(2, 3).::(1) = List(1, 2, 3)}}}
```
Comment (dwijnand) [inline src/library/scala/collection/immutable/List.scala:98]: Should this have used the `@example` tag?

**Comment by odersky** [comment]
I just added a comment to scala/scala-dev#584. I don't think dropping usecase is the answer as it fulfills a real need.

**Comment by SethTisue** [comment]
"fixes" https://github.com/scala/bug/issues/6071

**Pull request merged**